### PR TITLE
test(console): cover the operational views' failure paths

### PR DESCRIPTION
**Pull request by oxoxDev**
## Summary

Adds 59 unit files over the console's operational surface — the weakest chapter in the product for edge coverage, with several modules previously at zero tests. Each asserts what a view says when its fetch or action **fails**, rather than what it renders on the happy path.

Three real defects surfaced while writing them. None is fixed here: each is committed as a skipped test asserting the safe behaviour, so the bug is pinned rather than described.

## API Or Behavior Changes

None. No file under `frontend/src` is touched.

## Tests

59 new files, 174 passing, 3 skipped.

**The three skipped tests are findings, not gaps:**

- **`TaskCard` Resume double-fires.** `disabled={rows.length > 0}` is the only guard, and `rows` cannot change between two synchronous clicks, so a real double-click sends two re-dispatches — a second agent turn on work already re-queued. Fails 2 ≠ 1 against unmodified source.
- **Workspace rename is fire-and-forget.** The dialog does `void rename(...)` and closes; a slow first response can land after a newer second rename and win, discarding the operator's later intent.
- **`PeopleView` reports the wrong reason** when `/auth/me` fails to answer. The gate collapses "member" and "never answered" into one branch. The safe half holds — no control is offered either way — so this is a truthfulness bug, not an authority hole.

**On the authority axis, a correction worth recording.** Many of these cells were recorded as authority gaps. They are not: every task and chat write behind these controls is `ScopedCompany`/`CompanyAuth` on the host, not `AdminScopedCompany`, so there is no member/admin split to assert and the console correctly invents no gate. Forcing `useCanManage` to always-allow leaves the entire suite green — which is the honest confirmation that these were never authority cells. What the tests pin instead is the mirror property: no spurious gate, and the write really does land for a member. The two genuinely admin-gated surfaces, the chat daily-budget menu and the memory engine picker, do gate, and are now pinned.

Two fixtures needed an admin session after the skill write routes moved to the admin boundary — they had established no role at all, so they were asserting against a surface that no longer exists.

## Documentation

None. The axis correction above belongs in the coverage ledger rather than in a doc.

## Summary by CodeRabbit

- **Tests**
  - Expanded automated coverage across chat, tasks, workspace, inbox, memory, pages, people, skills, artifacts, and observatory views.
  - Added validation for role-based access, successful and failed actions, retry behavior, optimistic update rollback, loading states, and error messaging.
  - Added coverage for file uploads, approvals, reactions, budgets, task controls, workflow proposals, and concurrent operations.
  - Documented known edge cases and currently skipped scenarios for future fixes.

### CODE REVIEW

File: frontend/test/unit/cov-rest-skills-install-lifecycle.test.ts [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  uninstall?: () => Promise<void>;
+}): OpenCompanyClient {
+  const install = opts.install ?? (() => Promise.resolve(INSTALLED));
+  const uninstall = opts.uninstall ?? (() => Promise.resolve());
+  return {
+    scopeFor: () => "/api/v1/companies/acme",
+    get: (path: string) => {
+      if (path.includes("/skills/registry")) return Promise.resolve([REGISTRY]);
+      if (path.endsWith("/auth/me"))
+        return Promise.resolve({ id: "u1", email: "a@b.c", role: "admin", company: "acme", hasPassword: true });
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-skills-install-lifecycle.test.ts]: Confirmed valid — verified against current code: `install`/`uninstall` in src/server/ops/skills.rs are AdminScopedCompany, and SkillsView.tsx derives canManage from /auth/me and hides Install accordingly. The test docstring and fixture were stale relative to that (claimed ScopedCompany/no gate). Flipped the fixture to role: "member" and now assert Install is withheld; corrected the docstring. Fixed in 691a3a9e5.

File: frontend/test/unit/cov-rest-skills-install-lifecycle.test.ts [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    expect(container.textContent).toContain("Standup writer");
+    const uninstallButton = container.querySelector<HTMLButtonElement>(
+      '[data-testid="installed-card"] button[aria-label="Uninstall"]',
+    )!;
+    await act(async () => {
+      uninstallButton.click();
+    });
+
+    // Still on screen — the refusal must not have removed the card.
+    expect(container.textContent).toContain("Standup writer");
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-skills-install-lifecycle.test.ts:151]: Confirmed valid — SkillsView.uninstall() does call toast.error(e.message) on rejection, and the test never checked it. Added a sonner mock and asserted toast.error was called with the rejection message. Verified with a red-run: neutering the toast.error call in SkillsView.tsx made this assertion fail before the fix, and passes now. Fixed in 691a3a9e5. (Same commit also covers the coderabbit finding on line 142, same underlying gap.)

File: frontend/test/unit/cov-rest-artifacts-tab-edit.test.ts [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        },
+      ],
+      updatedAtMillis: 3000,
+    };
+    const client = {
+      scopeFor: () => "/api/v1/companies/acme",
+      // The background refresh after a save re-reads the list — by the time it
+      // lands, the host's own roster already agrees with what the append
+      // answered, exactly as it would once the write is durable.
+      get: () => Promise.resolve([afterSecondTabAppended]),
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-artifacts-tab-edit.test.ts]: Confirmed valid — the mock get() returned the post-append snapshot on the initial render too, so the tab already showed v3 before Save was ever clicked; the assertion would pass even if the save response were never applied. Now the first get() call returns the stale v1-only ARTIFACT and subsequent calls return the post-append state, plus an explicit pre-save check that v3 is absent. Verified with a red-run against both the save-response replace path and the post-save refresh path individually and together; the test only goes red when both are broken. Fixed in 691a3a9e5.

File: frontend/test/unit/cov-rest-inbox-isolation-and-fail.test.ts [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+      inboxMessages: vi.fn(() => Promise.reject(new Error("mailbox host unreachable"))),
+    } as unknown as OpenCompanyClient;
+
+    await show(client);
+
+    const errorPane = container.querySelector('[data-testid="inbox-messages-error"]');
+    expect(errorPane).not.toBeNull();
+    expect(errorPane?.textContent).toContain("mailbox host unreachable");
+    expect(container.querySelector('[data-testid="inbox-empty"]')).toBeNull();
+  });
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-inbox-isolation-and-fail.test.ts:127]: Confirmed valid, and fixed in both files — this one (cov-rest-inbox-isolation-and-fail.test.ts) and the named sibling cov-rest-inbox-view.test.ts#L107-117. Both now assert the "Try again" button is present in the error pane. Verified with a red-run: renaming the button label in InboxView.tsx made both assertions fail before the fix, and both pass now. Fixed in 691a3a9e5.

File: frontend/test/unit/cov-rest-memory-engine-probe-non-persistent.test.ts [hunk 5]
```diff
@@ -0,0 +1,123 @@
...
+    },
+  ],
+};
+
+function clientWith(probe: EngineProbe): { client: OpenCompanyClient; put: ReturnType<typeof vi.fn> } {
+  const put = vi.fn(() => Promise.reject(new Error("apply should not be called by a probe")));
+  const client = {
+    scopeFor: () => "/api/v1/company/acme",
+    get: vi.fn(() => Promise.resolve(STATE)),
+    post: vi.fn(() => Promise.resolve(probe)),
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-memory-engine-probe-non-persistent.test.ts]: Confirmed valid — neither test checked that the probe route was actually hit; both assertions (put not called, health text unchanged) would pass even if the Test connection click did nothing at all. Routed post() by path, following the pattern already used in cov-rest-memory-engine-auth.test.ts, and now assert the probe call fires exactly once with engine: "supermemory" in both tests. Verified with a red-run: skipping the testMemoryEngine call in EngineSection.tsx made both assertions fail before the fix, and pass now. Fixed in 691a3a9e5.

File: frontend/test/unit/cov-rest-people-me-fetch-failure.test.ts [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+});
+
+describe("PeopleView, /auth/me merely failing to answer (not a real member)", () => {
+  it("still offers no control — the safe half of fail-closed holds even on a network failure", async () => {
+    const client = clientAs((path) => {
+      if (path.endsWith("/auth/me")) {
+        return Promise.reject(new ApiError(0, "network_error", "cannot reach the company host"));
+      }
+      return Promise.reject(new Error(`unexpected GET ${path}`));
+    });
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-people-me-fetch-failure.test.ts]: Confirmed valid — the get callback was a plain arrow function, not observable, so this would pass identically if PeopleView stopped reading /auth/me entirely and just defaulted to no controls. Wrapped it in vi.fn() and asserted it was called with a path ending in /auth/me. Verified with a red-run: hardcoding the fetch result without calling client.get in PeopleView.tsx made this assertion fail before the fix, and pass now. Fixed in 691a3a9e5.

File: frontend/test/unit/cov-rest-skills-install-lifecycle.test.ts [hunk 7]
```diff
@@ -0,0 +1,143 @@
...
+  });
```
[the remark this answers is not in the document]
Comment (oxoxDev) [reply frontend/test/unit/cov-rest-skills-install-lifecycle.test.ts:153]: Confirmed valid — same gap as the codex finding on line 141 of this file (both point at the same missing assertion). Mocked sonner and asserted toast.error was called with the uninstall rejection message. Verified with a red-run: neutering the toast.error call in SkillsView.tsx made this assertion fail before the fix, and pass now. Fixed in 691a3a9e5, single change covers both threads.

**Comment by oxoxDev** [comment]
Addressed all 7 open review threads (2 codex P2, 5 CodeRabbit Minor) with real fixes rather than skip-with-reason, since every one of them was a variant of "this test runs the code path but doesn't assert the thing it claims to cover" — exactly the class of gap a coverage PR exists to close.

- **cov-rest-skills-install-lifecycle.test.ts** (codex, member-role fixture): the fixture used `role: "admin"` while claiming to test a plain member, and the docstring said install/uninstall were ungated `ScopedCompany`. Verified against current code that both routes are `AdminScopedCompany` and `SkillsView` gates `Install` on `canManage`. Flipped the fixture to `role: "member"`, now asserts `Install` is withheld, corrected the docstring.
- **cov-rest-skills-install-lifecycle.test.ts** (codex + CodeRabbit, uninstall toast — two threads, same gap): the refusal test only checked the card stayed on screen, never the toast `SkillsView.uninstall()` actually shows. Mocked `sonner`, asserted `toast.error` with the rejection message.
- **cov-rest-artifacts-tab-edit.test.ts** (CodeRabbit): the mock `get()` returned the post-append snapshot on the very first render, so the tab already showed v3 before Save was clicked — the assertion would've passed even with a broken save path. Seeded the initial render with the stale v1 copy, added a pre-save check that v3 is absent.
- **cov-rest-inbox-isolation-and-fail.test.ts + cov-rest-inbox-view.test.ts** (CodeRabbit, one thread naming both files): neither message-read-failure test asserted the "Try again" control despite claiming to. Added the assertion to both.
- **cov-rest-memory-engine-probe-non-persistent.test.ts** (CodeRabbit): both tests only checked `put` wasn't called and the screen was unchanged — true even if the probe click did nothing. Routed `post()` by path and asserted the probe call fires once with the selected candidate.
- **cov-rest-people-me-fetch-failure.test.ts** (CodeRabbit): the `get` callback wasn't wrapped in `vi.fn()`, so the test couldn't tell "PeopleView read /auth/me and it failed" apart from "PeopleView never asked." Wrapped it and asserted the call.

Nothing was skipped as stylistic — all 7 were real assertion gaps.

**Proof each fix actually bites**: for every strengthened assertion, I mutated the corresponding production code locally (disabled the toast call, the canManage gate, the probe request, the /auth/me read, the save's state-replace path) to simulate the exact regression the finding described, confirmed the test went red, then reverted and confirmed green again. Details and failure output are in the PR conversation for whoever wants to check the work; the working tree shows only the 6 test-file diffs (`git diff --stat`), no production files touched.

Gates: `npm run typecheck`, `typecheck:unit`, `typecheck:e2e`, `npx vitest run` (571 files / 4991 tests, all green), `npm run build` — all pass at head `691a3a9e5`.

@coderabbitai review
